### PR TITLE
github: update CODEOWNERS with Regen maintainers

### PR DESCRIPTION
**Pull request by erikgrinaker**
Is there anyone else who should be on here @robert-zaremba? Also, you may need to get added to the `regen_iavl_maintain` team for commit access.

### CODE REVIEW

**Comment by robert-zaremba** [comment]
Thanks for the PR. I will check with our team if someone else feels strong to be here as well.

**Comment by robert-zaremba** [comment]
@erikgrinaker please add also @amaurymartiny . Thanks!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,2 +1,2 @@
-# Everything goes through Erik & Aditya
-*     @erikgrinaker @AdityaSripal
+# Everything goes through Robert
+*     @robert-zaremba
```
Comment (robert-zaremba) [inline .github/CODEOWNERS:2]:
```suggestion
*     @robert-zaremba
*     @amaurymartiny 
```

**Review by robert-zaremba — APPROVED** [review]
Pre-approving - added request to add Amaury as well.

**Review by marbar3778 — APPROVED** [review]
🥳

**Pull request merged**